perf_bench_test.go

```diff
@@ -37,114 +41,192 @@ import (
 var (
 	isColdStart     = flag.Bool("coldStart", false, "Profile cold starts (default is false)")
 	vmNum           = flag.Int("vm", 2, "The number of VMs")
-	targetReqPerSec = flag.Int("requestPerSec", 2, "The target number of requests per second")
-	executionTime   = flag.Int("executionTime", 2, "The execution time of the benchmark in seconds")
+	targetReqPerSec = flag.Int("requestPerSec", 10, "The target number of requests per second")
+	executionTime   = flag.Int("executionTime", 1, "The execution time of the benchmark in seconds")
+	funcNames       = flag.String("funcNames", "helloworld", "Name of the functions to benchmark")
+	perfEvents      = flag.String("perfEvents", "", "Perf events (run `perf stat` if not empty)")
+
+	envPath = "PATH=" + os.Getenv("PATH")
+)
+
+const (
+	AveExecTime = "AveExecTime"
+	RealRPS     = "RealRPS"
 )
 
+// TODO: Change this from test function to normal function, and move to another file
+func TestBenchMultiVMRequestPerSecond(t *testing.T) {
+	log.SetLevel(log.InfoLevel)
+
+	for i := 4; i <= 4; i += 4 {
+		err := setupBenchRPS()
+		require.NoError(t, err, "Setup TestBenchRequestPerSecond error")
+
+		//Subtest
+		testName := fmt.Sprintf("Test_%dVM", i)
+		*targetReqPerSec = 10
+		*perfEvents = "instructions,LLC-loads,LLC-load-misses"
+		*vmNum = i
+		*funcNames = "rnn_serving"
+
+		testResult := t.Run(testName, TestBenchRequestPerSecond)
+		require.True(t, testResult)
+
+		err = teardownBenchRPS()
+		require.NoError(t, err, "Teardown TestBenchRequestPerSecond error")
+	}
+}
+
+func setupBenchRPS() error {
+	// sudo mkdir -m777 -p /tmp/ctrd-logs && sudo env "PATH=$PATH" /usr/local/bin/firecracker-containerd --config /etc/firecracker-containerd/config.toml 1>/tmp/ctrd-logs/fccd_orch_noupf_log_bench.out 2>/tmp/ctrd-logs/fccd_orch_noupf_log_bench.err &
+	var CTRDLOGDIR = "/tmp/ctrd-logs"
```

Comment:
a variable should not be named like that. Also, use const type for constants.

Reply:
Done